backend/core/src/main/java/org/sonarsource/sonarlint/core/repository/reporting/PreviouslyRaisedFindingsRepository.java

```diff
@@ -52,6 +53,22 @@ private static <F extends RaisedFindingDto> Map<URI, List<F>> addOrReplaceFindin
     return findingsPerFile;
   }
 
+  /**
+   * Increment the telemetry counter for issues that are AI fixable
+   * Avoid incrementing for already known issues, to have a more relevant counter
+   */
+  public void countAiFixableIssuesForTelemetry(String scopeId, Map<URI, List<RaisedIssueDto>> raisedFindings, TelemetryService telemetryService) {
+    var previousFindings = previouslyRaisedIssuesByScopeId.get(scopeId);
+    if (!previousFindings.isEmpty()) {
+      raisedFindings.forEach((uri, issues) -> {
+        var previousFindingsForFile = previousFindings.get(uri);
+        issues.stream()
+          .filter(i -> i.isAiCodeFixable() && !previousFindingsForFile.contains(i))
+          .forEach(i -> telemetryService.fixSuggestionApplicable());
```

Comment:
Same idea here, you introduce coupling between the repository, that should be something really simple, storing stuff in memory, and the telemetry service